### PR TITLE
refactor!: Untemplate `Vertex`

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR makes `Vertex` a concrete type, which is possible after moving input track to be a concrete type erased type.

Part of:
- #2842 

Blocked by:
- #2876

### CODE REVIEW

**Comment by paulgessinger** [comment]
This is ready to review now @andiwand.

File: Core/include/Acts/Vertexing/Vertex.ipp
Comment (andiwand) [inline Core/include/Acts/Vertexing/Vertex.ipp:1]: this can be a cpp file now?

Comment (paulgessinger) [reply Core/include/Acts/Vertexing/Vertex.ipp:1]: Yes! But I'd like to do this at the very end if possible.

**Review by andiwand — APPROVED** [review]
that's a blessing :pray:

**Comment by paulgessinger** [comment]
This is ready now @andiwand

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [dfd4356fcc2610b24fe4845531bd1045de82de8d]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/35829484) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6842368)!

**Pull request merged**